swirl_lm/base/driver.py

```diff
@@ -456,8 +456,38 @@ def step_fn(state):
   return strategy.run(step_fn, args=(init_state,))
 
 
+def strategy_and_coordinates(params: parameters_lib.SwirlLMParameters):
+  computation_shape = np.array([params.cx, params.cy, params.cz])
+  logging.info('Computation_shape is %s', str(computation_shape))
+  strategy = driver_tpu.initialize_tpu(
+      tpu_address=FLAGS.target, computation_shape=computation_shape)
+  num_replicas = strategy.num_replicas_in_sync
+  logging.info('TPU is initialized. Number of replica is %d', num_replicas)
+  logical_coordinates = tpu_util.grid_coordinates(computation_shape).tolist()
+  return strategy, logical_coordinates
+
+
+def get_init_state(
+  customized_init_fn: Union[types.InitFn, Any],
+  strategy,
+  params,
+  logical_coordinates,
+):
+  t_start = time.time()
+  init_fn = _init_fn(params, customized_init_fn)
+  state = driver_tpu.distribute_values(
+      strategy, value_fn=init_fn,
```

Comment:
In my previous version, I had to remove the tf.function-wrapper here.

Reply:
We added that wrapper recently to be able to run very large scale simulations, so I think we'll need to do something about it in your version, too. Yi-fan knows the intricacies of the wrapper change so I'll chat with him to figure out how we should go about it. It's OK not to have the wrapper for now.